dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java

```diff
@@ -234,7 +234,10 @@ public Token addQuayToken(@ApiParam(hidden = true) @Parameter(hidden = true, nam
     private void checkIfAccountHasBeenLinked(String username, TokenType tokenType) {
         Token existingToken = tokenDAO.findTokenByUserNameAndTokenSource(username, tokenType);
         if (existingToken != null) {
-            String msg = "A '" + tokenType.toString() + "' token already exists on Dockstore for the account '" + username + "'";
+            User dockstoreUser = userDAO.findById(existingToken.getUserId());
+            String msg = "The '" + tokenType.toString() + "' account '" + username
```

Comment:
I'd suggest an action as well in the message, e.g., roughly, "...is already linked to another Dockstore account with the name jdoe. Please log-in to Dockstore with that account". Bonus if you can identify the type and name of account, e.g., log-in to Dockstore with your jdoe GitHub account.

Also, your code is fine, but FYI you can use MessageFormat to create strings constructed from a bunch of variables . I personally find it more readable, but YMMV.

Reply:
In the current state, the message will say something like:
 `The 'github.com' account 'myGithubAccount' is already linked to another Dockstore account with the name 'myDockstoreAccount'.`

I'm thinking modifying to:
`The 'github.com' account 'myGithubAccount' is already linked to another Dockstore account with the name 'myDockstoreAccount'. You can login to Dockstore using your 'myGithubAccount' 'github.com' account.`

Thoughts? I can see how this might be hard to read in a snackbar component... perhaps its worth extending the duration on the popup before it disappears?

Reply:
1. Doesn't this show up in the big red box as well? If so, I don't think you need to increase the duration
2. You could word slightly more concisely: The "github.com" user "myGithubAccount" is already linked to the Dockstore user "myDockstoreAccount". Login to Dockstore using your "github.com" "myGithubAccount" user.

Also, just noticed this, but if you're going to use quotes in the sentence, I think you should use double quotes instead of single quotes, English grammar-wise.

Reply:
The big red box only pops up when linking with Google, all the other link buttons are attached to the snackbar. See [this](https://github.com/dockstore/dockstore-ui2/blob/11ca91a3d79a0ff5395ed744f2db7bc60b335023/src/app/loginComponents/accounts/external/accounts.service.ts#L54).

I'll update the text to use double quotes instead of single quotes and reword.